---
title: "Known markers for cell-type identification"
author: "Evelin Aasna"
date: "7/15/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
library(Seurat)
library(ggplot2)
load(file = "/icgc/dkfzlsdf/analysis/B210/Evelin/seurat_object/integrated_seu.RData")
source("/icgc/dkfzlsdf/analysis/B210/Evelin/git-repo/marker_library.R")
```


Ideas for violin plotting??
1. visualize cluster marker genes
2. visualize highly differentially expressed genes
3. visualize known markers for specific cell types


# Top 10 variable features for each assay
```{r}
find_clusters <- function(seu, type, assay, nPCA){
  DefaultAssay(seu) <- assay
  seu <- FindNeighbors(seu, dims = 1:nPCA, reduction = paste0(type, ".pca"), assay = assay)
  seu <- FindClusters(seu, verbose = FALSE)
  return(seu)
}
```


## Log normalized counts
```{r}
type = "log"
assay = "RNA"
nPCA = 20
seu = find_clusters(seu, type, assay, nPCA)
```

```{r fig.width=5, fig.height=5}
titlesize = 15
DimPlot( seu, reduction = "log.umap" ) + 
  labs(title = type) +
  theme(plot.title = element_text(size = titlesize))
```

```{r fig.width=10, fig.height=25}
VlnPlot(seu, features = VariableFeatures(seu, assay = "RNA")[1:10], pt.size = 0.2, ncol = 2)
```

Many genes are variably expressed in cluster 12
```{r}
markers12 = row.names(FindMarkers(seu, ident.1 = 12, min.pct=0.25, only.pos = TRUE)[1:40,])
```

Which celltype markers are variably expressed in cluster 12
```{r}
markers12[which(markers12 %in% unlist(marker.list))]
unlist(marker.list)[which(unlist(marker.list) %in% c("ACTA2", "MCAM", "MGP"))]
```

Cluster 12 is very small (2% of all cells)
```{r}
length(Idents(seu)[which(Idents(seu)==12)])
```


## SCTransformed counts
```{r}
type = "SCT"
assay = "SCT"
nPCA = 30
seu = find_clusters(seu, type, assay, nPCA)
```

```{r fig.width=5, fig.height=5}
titlesize = 15
DimPlot( seu, reduction = "log.umap" ) + 
  labs(title = type) +
  theme(plot.title = element_text(size = titlesize))
```

```{r fig.width=10, fig.height=25}
VlnPlot(seu, features = VariableFeatures(seu, assay = "SCT")[1:10], pt.size = 0.2, ncol = 2)
```

Cluster 5 seems to differ
```{r}
markers5 = row.names(FindMarkers(seu, ident.1 = 5, min.pct=0.25, only.pos = TRUE)[1:40,])
```

Which celltype markers are variably expressed in cluster 5
```{r}
unlist(marker.list)[which(unlist(marker.list) %in% markers5[which(markers5 %in% unlist(marker.list))])]
```

Cluster 5 size (%)
```{r}
length(Idents(seu)[which(Idents(seu)==5)])/length(Idents(seu)) * 100
```


```{r}
find_celltype_markers <- function(ident1, ident2){
  markers = row.names(FindMarkers(seu, ident.1 = ident1, ident.2 = ident2, min.pct=0.25, only.pos = TRUE)[1:40,])
  print(unlist(marker.list)[which(unlist(marker.list) %in% markers[which(markers5 %in% unlist(marker.list))])])
  print(length(Idents(seu)[which(Idents(seu)==ident1)])/length(Idents(seu)) * 100)
  return(markers)
}
```

```{r}
markers9 = find_celltype_markers(9, NULL)
```
```{r}
markers1 = find_celltype_markers(1, NULL)
```


```{r}
markers1_11 = find_celltype_markers(1, 11)
```

```{r}
markers11_1 = find_celltype_markers(11, 1)
```

```{r}
markers8 = find_celltype_markers(8, NULL)
```

## CCA integrated + SCTransformed counts
```{r}
type = "integrated"
assay = "integrated"
nPCA = 30
seu = find_clusters(seu, type, assay, nPCA)
```


```{r}
DimHeatmap(seu, dims = c(1,2), cells = 500, assays = "integrated", balanced = TRUE )
```

```{r fig.width=5, fig.height=5}
titlesize = 15
DimPlot( seu, reduction = "log.umap" ) + 
  labs(title = type) +
  theme(plot.title = element_text(size = titlesize))
```

```{r fig.width=10, fig.height=25}
DefaultAssay(seu) <- "integrated"
VlnPlot(seu, features = VariableFeatures(seu, assay = "integrated")[1:10], pt.size = 0.2, ncol = 2)
```

```{r}
find_celltype_markers(1, 3)
```



```{r fig.width=10, fig.height=25}
FeaturePlot(seu, reduction = "umap", features = c("PAEP", "CCL20", "CXCL8", "HSPA6", "MMP3", "SPP1", "MMP1", "MMP10", "IGFBP1", "G0S2" ), 
    pt.size = 0.2, ncol = 2)
```


# Known markers from cell atlas of human uterus
High levels of gene expression
Endometrial epithelial: KRT8, KRT18, EPCAM and CLDN3
Endometrial stromal: MME, FN1, COL3A1, HOXA10
Endothelial: CD34, VWF, CLDN5, SOX18
SMA+(smooth muscle and myofibrolblasts): ACTA2, MYH11, MYL6, MYL9
Endometrial immune: PTPRC, CD68, CD163 and CD96


# Known markers from molecular and cellular cartography
Heterogeneous population of fibroblasts: express matrix protein genes (e.g. PCOLCE, COL6A2) and is negative for PAEP, CD45, and CD31

```{r}
c("PCOLCE", "COL6A2")
```

# Known markers from hormone responsive organoid cultures
Markers for putative stem cells: SSEA-1+(gene FUT4), SOX9+ 
```{r}
c("FUT4", "SOX9")
```

